osu.Game/Graphics/UserInterfaceV2/DirectorySelector.cs

```diff
@@ -90,6 +92,15 @@ private void updateDisplay(ValueChangedEvent<DirectoryInfo> directory)
                     foreach (var drive in drives)
                         directoryFlow.Add(new DirectoryPiece(drive.RootDirectory));
                 }
+                else if (!directory.NewValue.Exists)
```

Comment:
if you follow my structure, you can write this without the initial check. it can just run the `while` loop regardless.